### DISCUSSION TITLE
Calling from_dict for multiple sources - multiple steps...

### DISCUSSION THREAD

**Discussion by MartyBolton** [bmsan/DictGest#28] [Q&A] [answered]
Hello I was talking to you on Reddit about your library - we are using this in our project! Love what you are doing here.
Question about how to structure populating our Python "view models" from multiple sources. 
Can you give us some guidance on the following:

What is best way of having developers populate a View Model or data-class if there are many sections and also each section may have database views ready and some may need mockup data. Would you recommend trying to combine all the dict data as a super_dict object and calling from_dict() once or is it okay to have each section define its own route and call from_dict multiple times for each sub-element of a dataclass?

For example lets say you have
class Report:
    Heading : ...
    Detail1 : ...
    Detail2 : ...
    Detail3: ...
   Footer: ...

Lets say you need to chop this up into sprints where a developer can populate these from mockup data and real data as the project goes along - whats the best way to do this? Would you make a single call or multiple calls and then combine them?

thanks
Marty

**Comment by bmsan** [comment, marked as answer]
Sorry for the very late reply. 
Somehow I missed the notification for this message and discovered it by chance now.

You could definitely populate each member field individually calling from_dict for each of them (some would be populated with real data and some with mockup data), and at the end just add them to the Report class. 
 
Another approach: you could insert into your raw input dict structure a "mockup_detail1_field" which is of type Detail1.

```
mockup_detail1 = Detail1Type(...placeholder_values...)

input_data["mockup_detail1_field"] = mockup_detail1

Report: Route(...., detail1="mockup_detail1_field", ....) 
```

Since mockup_detail1_field is of the right type dictgest would "leave it alone".

Later you could remove the mockup and define the real route/path for detail1.

**Reply by MartyBolton** [reply]
Thank you!
